### PR TITLE
Fix Dante's review comments.

### PR DESCRIPTION
**Pull request by navilan**
- Clean up gemspec file.
- Remove the autogenerated .rspec file.
- Fix JS defaults issue.
- Make rspec consistently use expect convention.
- lml/unconfirm#1: Removed all but one puts statement.
- Make sure rspec tests actually pass.

TODO:
- [x] Use `cache` instead of session.

### CODE REVIEW

**Comment by navilan** [comment]
@Dantemss / @jpslav  - This one is ready for review.

File: unconfirm.gemspec [hunk 1]
```diff
@@ -17,9 +17,8 @@ Gem::Specification.new do |s|
                           future attempts at the same action.}
   s.homepage    = "https://github.com/lml/unconfirm"
   s.license       = "MIT"
-  s.files         = `git ls-files -z`.split("\x0")
-  s.executables   = s.files.grep(%r{^bin/}) { |f| File.basename(f) }
-  s.test_files    = s.files.grep(%r{^(test|spec|features)/})
+  s.files         = Dir["{app,config.db.lib}/**/*"] + ["Rakefile", "LICENSE", "README.md", "CONTRIBUTING.md"]
```
Comment (Dantemss) [inline unconfirm.gemspec]: I think this is supposed to be `app,config,db,lib` (commas, not periods)

**Comment by Dantemss** [comment]
Looks good except for 1 small thing in the gemspec

File: unconfirm.gemspec (hunk 1 above)
Comment (navilan) [inline unconfirm.gemspec]: yikes.  Not sure what I did there.  Will fix. Thanks @Dantemss .

Comment (jpslav) [inline unconfirm.gemspec]: Wow.  You have good vision @Dantemss!

Comment (navilan) [inline unconfirm.gemspec]: Yep :) I should probably add spaces after the commas for people with broken eyes like me :)

**Comment by jpslav** [comment]
@lakshmivyas -- want to merge this?

**Comment by navilan** [comment]
@jpslav - yes, Thank you :)  Need @Dantemss' eyes on the final commit.

File: app/models/unconfirm/user_settings.rb [hunk 2]
```diff
@@ -25,9 +37,20 @@ def self.table_name_prefix
       'unconfirm_'
     end
 
-    def self.for (user)
+    def self.for(user)
+      if user
+          find_or_create_by_user_id(:user_id => user.id)
```
Comment (Dantemss) [inline app/models/unconfirm/user_settings.rb:42]: This method might be deprecated in Rails 4. Not an issue now, but good to know for the future.

**Comment by Dantemss** [comment]
Looks good to me.

**Comment by navilan** [comment]
@Dantemss - Thank you.  Will fix the `find_or_create_by_` as part of the next commit.

**Comment by Dantemss** [comment]
OBE probably

**Pull request closed without merging**